common/tools/dev-tool/.eslintrc.json

```diff
@@ -0,0 +1,14 @@
+{
+  "parser": "@typescript-eslint/parser",
+  "plugins": ["@typescript-eslint"],
+  "extends": [
+    "eslint:recommended",
+    "plugin:@typescript-eslint/eslint-recommended",
+    "plugin:@typescript-eslint/recommended"
+  ],
+  "rules": {
+    "@typescript-eslint/explicit-function-return-type": "off",
+    "@typescript-eslint/no-explicit-any": "off",
+    "@typescript-eslint/no-non-null-assertion": "off"
```

Comment:
I'm curious how these rules impacted you

Reply:
For explicit any it was just that I used it a lot for objects like package json, but I've gone through and annotated where I used it. In `printer.ts` since there are a lot of functions that use `(...args: any[]) => T` signatures I disabled it on the whole file, and I just disabled it on the line where it's used for the type that results from importing a package.json.

We do have all three of these disabled in some fashion in the SDK package linter configs.

`explicit-function-return-type` is just a rule that I really don't like personally. I think it's excessive in a lot of cases to have to explicitly specify a function return type if it can be inferred. For example in an expression like `() => error.stack.shift()` this should just be inferrable. I don't think that writing out `() : NodeJS.CallSite | undefined => error.stack.shift()` really adds anything to the code (if anything I think it detracts from the readability), but the rule really wants that annotation there even when I turn on `allowExpressions` in the rule settings.

Reply:
I think the rule exists because sometimes (especially with public surface) it's possible to mutate the return shape unintentionally in ways that are hard to reason about. E.g. your return becomes `any` and nothing breaks until you run it.